catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/IdentificationPlugin.java

```diff
@@ -226,25 +319,49 @@ private String getRegistryId(Metacard mcard) {
                 .toString();
     }
 
-    public void init() throws FederationAdminException {
-
-        List<Metacard> registryMetacards = federationAdmin.getRegistryMetacards();
-
+    public void init()
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+
+        List<Metacard> registryMetacards;
+        Filter registryFilter = filterBuilder.attribute(Metacard.TAGS)
+                .is()
+                .like()
+                .text(RegistryConstants.REGISTRY_TAG);
+        Subject systemSubject = Security.getInstance()
+                .getSystemSubject();
```

Comment:
I needed access to the Impl method setPageSize, which means I would cast that query on the next line...

`Query query = new QueryImpl(registryFilter);
((QueryImpl) query).setPageSize(1000);
QueryRequest request = new QueryRequestImpl(query);`

I can change it to this ^ if that is better. What do you think?